### PR TITLE
Expose Shutdown func for EventBroadcaster

### LINKED ISSUE #83487: record.NewBroadcaster leaks goroutines
In order to initialize a client-go `record.EventRecorder`, you need an `record.EventBroadcaster`.  `record.NewBroadcaster` and friends initialize a new `watch.Broadcaster`:

https://github.com/kubernetes/kubernetes/blob/3317805652d145c4003f5c534c7d87fb5cd87d50/staging/src/k8s.io/client-go/tools/record/event.go#L133-L137

`watch.NewBroadcaster` launches a goroutine:

https://github.com/kubernetes/kubernetes/blob/3317805652d145c4003f5c534c7d87fb5cd87d50/staging/src/k8s.io/apimachinery/pkg/watch/mux.go#L67-L77

This goroutine is stoppable with  `watch.Broadcaster.Shutdown`.  However, the `record.EventBroadcaster` interface returned above does not expose anything that calls that shutdown method, AFAICT.  This means that that goroutine cannot ever be cleaned up.

That's not great.  It's fine if you're using in like k/k does and never stopping anything, but if you ever want to stop your controllers and start up new ones, it becomes an issue.

Tested/tracked down against master (as of 3317805652d145c4003f5c534c7d87fb5cd87d50).

/sig apimachinery
/kind bug

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
this PR exposes Shutdown func for EventBroadcaster.

**Which issue(s) this PR fixes**:
Fixes #83487

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by SataQiu — COMMENTED** [review]
Thanks @tedyu 
/lgtm

**Review by yastij — COMMENTED** [review]
/lgtm
/approve 
/assign @wojtek-t 

however should also be fixed when using `events.k8s.io`

**Comment by yastij** [comment]
@tedyu - can you handle this for the `events/` package too ?

**Comment by tedyu** [comment]
Change to events/ package has been added.

**Comment by yastij** [comment]
/approve

@tedyu - can you fix the lint issues ?

File: staging/src/k8s.io/client-go/tools/events/event_broadcaster.go [hunk 1]
```diff
@@ -102,16 +102,20 @@ func newBroadcaster(sink EventSink, sleepDuration time.Duration, eventCache map[
 	}
 }
 
+func (eventBroadcaster *eventBroadcasterImpl) Shutdown() {
+	eventBroadcaster.Broadcaster.Shutdown()
+}
+
 // refreshExistingEventSeries refresh events TTL
-func (e *eventBroadcasterImpl) refreshExistingEventSeries() {
+func (eventBroadcaster *eventBroadcasterImpl) refreshExistingEventSeries() {
```
Comment (mengqiy) [inline staging/src/k8s.io/client-go/tools/events/event_broadcaster.go]: IIRC the convention in golang is to use a short name for receivers.

Comment (tedyu) [reply staging/src/k8s.io/client-go/tools/events/event_broadcaster.go]: I was trying to make kubernetes-verify job happy.

Comment (wojtek-t) [inline staging/src/k8s.io/client-go/tools/events/event_broadcaster.go]: If you want to change names, please do that in a separate commit.
[That said, I actually think that "e" was better than such a long name.

Comment (tedyu) [reply staging/src/k8s.io/client-go/tools/events/event_broadcaster.go]: @wojtek-t 
I have taken out the name change.
Please check again.

File: staging/src/k8s.io/client-go/tools/record/event.go [hunk 2]
```diff
@@ -169,6 +172,10 @@ func (eventBroadcaster *eventBroadcasterImpl) StartRecordingToSink(sink EventSin
 		})
 }
 
+func (eventBroadcaster *eventBroadcasterImpl) Shutdown() {
```
Comment (yastij) [inline staging/src/k8s.io/client-go/tools/record/event.go]: this should be `e` instead of `eventBroadcaster `

Comment (tedyu) [reply staging/src/k8s.io/client-go/tools/record/event.go]: @yastij 
Your comment has been addressed.
